### PR TITLE
Be 85 create service for trainee staatus changes

### PR DESCRIPTION
**Pull request by nidaffs**
Add TraineeStatusService with methods which allow to change status trainee

### CODE REVIEW

**Review by nazfurdychka — APPROVED** [review]

File: src/main/java/com/exadel/project/trainee/service/TraineeStatusService.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    public void changeTraineeStatus(AdditionalInfo additionalInfo) {
+        TraineeStatus status = additionalInfo.getTraineeStatus();
+        switch (status) {
+            case EMAIL_NOT_CONFIRM:
+                additionalInfo.setTraineeStatus(TraineeStatus.REGISTERED);
+                break;
+            case REGISTERED:
+                additionalInfo.setTraineeStatus(TraineeStatus.RECRUITER_INTERVIEW_PENDING);
+                break;
+            case RECRUITER_INTERVIEW_ACCEPTED:
```
Comment (serkushner) [inline src/main/java/com/exadel/project/trainee/service/TraineeStatusService.java:31]: ?

File: src/main/java/com/exadel/project/trainee/service/TraineeStatusService.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+                if (isApproved) {
+                    additionalInfo.setTraineeStatus(TraineeStatus.TECHNICAL_INTERVIEW_ACCEPTED);
+                } else {
+                    additionalInfo.setTraineeStatus(TraineeStatus.TECHNICAL_INTERVIEW_REJECTED);
+                }
+                break;
+            default:
+                throw new TraineeStatusIsNotAvailableForChangesException();
+        }
+        additionalInfoRepository.save(additionalInfo);
```
Comment (serkushner) [inline src/main/java/com/exadel/project/trainee/service/TraineeStatusService.java:67]: so difficult to understand such structures, pls find a way how we can avoid using switch

**Comment by serkushner** [comment]
So try to avoid if or switch case with polymorphism

**Review by serkushner — CHANGES_REQUESTED** [review]
So try to avoid if or switch case, replace with polymorphism

**Pull request closed without merging**